test/config-luatest/instance_config_schema_test.lua

```diff
@@ -43,7 +43,7 @@ local function validate_fields(config, record)
                 validate_fields(v1, v.items)
             end
         end
-        if v.type ~= 'record' or not v.enterprise_edition or is_enterprise then
+        if not v.enterprise_edition or is_enterprise then
```

Comment:
The removed condition actually looks as a mistake: we should check that the keys are the same disregarding of the type of the value. I see that it is needed in `test_wal` below, since the `wal` section receives a first non-record enterprise-only option.

However, I would like to re-check it with Mergen.

@ImeevMA Do you confirm that this change is valid?

Reply:
It looks like it was most likely a mistake. I seem to have added this condition thinking `enterprise_edition` could only be `true` for `record`.